### PR TITLE
Modernize actions CI

### PR DESCRIPTION
**Pull request by chapulina**
Update to new style. Focal support will be added to Citadel.

### CODE REVIEW

**Comment by chapulina** [comment]
The CMake warning on Ubuntu is not introduced by this PR, it's showing now because of the new cmake warnings plugin on Jenkins. Here's an issue about it: https://github.com/ignitionrobotics/ign-tools/issues/23

File: .github/ci/packages.apt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+libignition-physics-dev
+libignition-plugin-dev
+libignition-sensors2-dev
+libignition-tools-dev
+libignition-transport7-dev
+libsdformat8-dev
+libtinyxml2-dev
+libwebsockets-dev
+libyaml-dev
+python-yaml
```
Comment (nkoenig) [inline .github/ci/packages.apt:19]: Why is this needed?

Comment (chapulina) [reply .github/ci/packages.apt:19]: No idea, it was just propagated from bitbucket pipelines:

https://github.com/ignitionrobotics/ign-launch/blob/a284b73fae4c116529295ac2a84a990dfb405c1a/bitbucket-pipelines.yml#L17

I'll remove it and see what happens.

Comment (chapulina) [reply .github/ci/packages.apt]: It works without it: d85eb63

**Comment by chapulina** [comment]
> is there anywhere else I should look to verify this PR for future reference?

Since this PR is only touching the actions, I think it should be enough to check that the file changes look reasonable and the actions are passing.

When I merge it forward to Citadel, I'll add Focal CI, which hopefully will pass without the need for extra changes. But if there are more changes needed, we should also make sure that they look reasonable and don't break the rest of CI, or behaviour, API, ABI, etc.
